### ISSUE TITLE
Uncaught (in promise) TypeError: url_1.URL is not a constructor

### ISSUE THREAD

**Issue by acai2046** [redis/node-redis#2420]
### Description

```
const client = redis.createClient({
		url: 'redis://127.0.0.1:6379'
	});
```
--------------
Uncaught (in promise) TypeError: url_1.URL is not a constructor

### Node.js Version

v14.0.0

### Redis Server Version

redis@4.6.4

### Node Redis Version

`-- redis@4.6.4

### Platform

windows

### Logs

```bash
Uncaught (in promise) TypeError: url_1.URL is not a constructor
    parseURL chunk-vendors.js:53223
    _RedisClient_initiateOptions chunk-vendors.js:53264
    RedisClient chunk-vendors.js:52563
    _createSuperInternal chunk-vendors.js:29839
    Commander chunk-vendors.js:74825
    create chunk-vendors.js:53217
    createClient chunk-vendors.js:45604
    _callee$ redis.js:8
    tryCatch chunk-vendors.js:56382
    makeInvokeMethod chunk-vendors.js:56463
    defineIteratorMethods chunk-vendors.js:56407
    asyncGeneratorStep chunk-vendors.js:82818
    _next chunk-vendors.js:82837
    _asyncToGenerator chunk-vendors.js:82842
    _asyncToGenerator chunk-vendors.js:82834
    redisCreateClient redis.js:3
    _callee$ main.js:6
    tryCatch chunk-vendors.js:81473
    makeInvokeMethod chunk-vendors.js:81554
    defineIteratorMethods chunk-vendors.js:81498
    asyncGeneratorStep chunk-vendors.js:26629
    _next chunk-vendors.js:26648
    _asyncToGenerator chunk-vendors.js:26653
    _asyncToGenerator chunk-vendors.js:26645
    test main.js:5
    <anonymous> main.js:12
    7RBU index.js:1127
    __webpack_require__ index.js:854
    fn index.js:151
    1 index.js:1031
    __webpack_require__ index.js:854
    checkDeferredModules index.js:46
    <anonymous> index.js:994
    <anonymous> index.js:997
```

**Comment by leibale** [comment]
Are you trying to import the package in the client side?

**Comment by acai2046** [comment]
With introduction

**Comment by leibale** [comment]
@acai2046 what do you mean by "With introduction"? Can you please share a little bit more? Thanks

**Comment by AlokBisht** [comment]
I am facing the same issue. when using redis with the nextjs project. however it works fine in a demo node app. any suggestions why the "URL" is undefined with nextjs app?

**Comment by leibale** [comment]
@AlokBisht make sure not to use the redis client from the client side, it only works on the server side (node, deno, ...)

**Comment by AlokBisht** [comment]
yeah, it's not on client side. like so many other reported issue, it seems there is some kind of type conflict happening due to some dependency in my project. that's my best guess after spending hours on this issue.

I tried using "ioredis" as well. it works but I see "TypeError: (0 , tls_1.connect) is not a function". so there is some dependency arrangement which is causing the types to be lost.

**Comment by leibale** [comment]
@AlokBisht I don't think that it's a type error, feels like there is actual code that is missing.. can you please create a repo that reproduces the issue so I'll be able to debug it?

**Comment by AlokBisht** [comment]
OK.. but my case seems similar to
https://stackoverflow.com/questions/67709483/sveltekit-typeerror-url-is-not-a-constructor
https://stackoverflow.com/questions/65399106/error-network-error-url-1-url-is-not-a-constructor-at-new-apolloerror

**Comment by AlokBisht** [comment]
My issue is resolved now. you were right @leibale  the redis code was reaching client side from one of the wrong reference in code. hence was getting the error. it works fine after I removed the wrong code ref.

**Comment by Tankonyako** [comment]
Have same, and this code working only on server side, and problem is 
<img width="350" alt="image" src="https://github.com/redis/node-redis/assets/48120579/0fc61b94-18b8-4a27-b0e5-f2472a46cc94">
require('url') gives Url not URL
